kernel/src/log_segment.rs

```diff
@@ -104,13 +113,240 @@ impl LogSegment {
     }
 }
 
+pub(crate) struct LogSegmentBuilder<'a> {
+    fs_client: &'a dyn FileSystemClient,
+    log_root: &'a Url,
+    checkpoint: Option<CheckpointMetadata>,
+    start_version: Option<Version>,
+    end_version: Option<Version>,
+    no_checkpoint_files: bool,
+    in_order_commit_files: bool,
+}
+impl<'a> LogSegmentBuilder<'a> {
+    pub(crate) fn new(fs_client: &'a dyn FileSystemClient, log_root: &'a Url) -> Self {
+        LogSegmentBuilder {
+            fs_client,
+            log_root,
+            checkpoint: None,
+            start_version: None,
+            end_version: None,
+            no_checkpoint_files: false,
+            in_order_commit_files: false,
```

Comment:
do we ever want non-ordered commit files?

Reply:
So this is a bit of a naming issue. The default behaviour for snapshot is to return commit files in reverse order. I made this the default for `LogSegment`.  Maybe I should instead make it `reverse_commit_files` and make `Snapshot` explicitly ask for reverse.

Reply:
I went ahead and switched it to `with_reversed_commit_files` PTAL.